client/app/scripts/liveblog-themes/module.js

```diff
@@ -115,53 +117,57 @@ import listTpl from 'scripts/liveblog-themes/views/list.ng1';
         }
         function loadThemes() {
             // load only global preference for themes.
-            api.global_preferences.query({where: {key: 'theme'}}).then((globalPreferences) => {
-                self.globalTheme = _.find(globalPreferences._items, (item) => item.key === 'theme');
+            const globalPrefPromise = api.global_preferences.query({where: {key: 'theme'}})
+                .then((globalPreferences) => {
+                    self.globalTheme = _.find(globalPreferences._items, (item) => item.key === 'theme');
+                });
+
+            // get all blogs
+            const blogsPromise = api.blogs.query().then((data) => {
```

Comment:
wouldn't this fetch like 25 results by default?

Reply:
Yes. It fetches all the blogs once and then filters them for each theme in this section: 

```
themes.forEach((theme) => {
    const matchingBlogs = self.blogs.filter((blog) =>
        blog.blog_preferences
        && blog.blog_preferences.theme === theme.name);

    theme.blogs_count = matchingBlogs.length;
    theme.blogs = matchingBlogs;
```

The criteria we were using before wasn't working even if you tried it on postman and would always returned 0 for each theme. 

At the moment I'm trying to figure why that is and I suspect it is because on the `blog_schema` the mapping for `blog_preferences` is set to False so the `blog_preferences.theme` is not queried against search operations. So I'm looking into it as a separate issue considering `blog_preferences.theme` is used in multiple places.